http/cookies.lua

```diff
@@ -0,0 +1,324 @@
+local http_patts = require "lpeg_patterns.http"
+local util = require "http.util"
+local psl = require "psl"
+
+local function parse_set_cookie(text_cookie, request, time)
+	local domain = request.headers:get("host")
+	local key, value, matched_cookie = assert(http_patts.Set_Cookie:match(
+		text_cookie))
+	-- allow custom time for specs to not error
+	time = time or os.time()
+	local cookie = {
```

Comment:
One intention of `http_patts.Set_Cookie` was that the output would be usable as a cookie directly.

To set defaults, use a metatable with __index.

Reply:
> 19:40 <~daurnimator> ryan: right. but you can modify the table returned by the pattern
19:42 <~daurnimator> though I guess you might not want to do that
19:42 <~daurnimator> as the pattern can return arbitrary keys
19:42 <~daurnimator> so maybe it's fine as it is...
19:44 <~ryan> another issue is, converting it just ended up as the same line count because most of the items have defaults
19:44 <~ryan> plus the fact that there's so many different naming conventions
19:45 <~ryan> httponly, ["max-age"], same_site
19:46 <~daurnimator> ryan: I was more thinking about saving the extra allocation
19:46 <~ryan> yeah.
19:46 <~daurnimator> but yeah; in this case not worth it
19:46 <~ryan> i think the way i did it is better though
19:47 <~ryan> most of the keys are new keys, so doing it by appending on the old table means reallocating a lot
19:47 <~ryan> the new table means that all but one field are preallocated
19:48 <~ryan> which i actually just fixed to make it so all fields are prealloced